libraries/botbuilder-dialogs/src/skillDialog.ts

```diff
@@ -0,0 +1,135 @@
+/**
+ * @module botbuilder-dialogs
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { BotFrameworkSkill } from 'botbuilder'; // bad import
+import { Activity, ActivityTypes, ConversationState, StatePropertyAccessor, TurnContext } from 'botbuilder-core';
+import { Dialog, DialogTurnResult } from './dialog';
+import { DialogContext } from './dialogContext';
+import { SkillDialogArgs } from './skillDialogArgs';
+import { SkillDialogOptions } from './skillDialogOptions';
+
+export class SkillDialog extends Dialog {
+    private readonly _activeSkillProperty: StatePropertyAccessor;
+    private readonly _conversationState: ConversationState; // Why is this restricted to ConversationState?
+    private readonly _dialogOptions: SkillDialogOptions;
+
+    /**
+     * A sample dialog that can wrap remote calls to a skill.
+     * 
+     * @remarks
+     * The options parameter in `beginDialog()` must be a `SkillDialogArgs` object with the initial parameters
+     * for the dialog.
+     * 
+     * @param SkillDialogOptions 
+     * @param dialogOptions 
+     * @param ConversationState 
+     * @param conversationState 
+     */
+    public constructor(dialogOptions: SkillDialogOptions, conversationState: ConversationState) {
+        super('SkillDialog'); // What about Dialog Id?
```

Comment:
@gabog just a nit, should we make `dialogId` configurable via `SkillDialogOptions`?